### PR TITLE
Add Czech translation

### PR DESCRIPTION
**Pull request by ceskyDJ**
Found problems:
- "DIRTY" - What does it mean?
- Missing translations for "VIEW LICENSES" and "CLOSE" (in app's info modal)
- "Invertible" (in "Image options") - I don't see it in app on my tablet. Could you tell me, what does it do, please?

### CODE REVIEW

**Comment by ceskyDJ** [comment]
Hi! I have found some problems while translating your app I need to help with. Two terms isn't displayed in my app version (Samsung Galaxy Tab S8, v0.5.3 Google Play 5030) and I don't know what them mean. There are two buttons without translation available in app's info (shown by tapping on app version in settings).

**Comment by ceskyDJ** [comment]
Some translations are a bit more further from your English original. In Czech native speak it doesn't make sense to "let it be and just translate it word-by-word", so I hope you are ok with it.

**Comment by adil192** [comment]
> * "DIRTY" - What does it mean?

Dirty builds have commits that are ahead of the latest release. It's similar to the definition of "dirty" in https://mirrors.edge.kernel.org/pub/software/scm/git/docs/gitglossary.html#def_dirty

> * Missing translations for "VIEW LICENSES" and "CLOSE" (in app's info modal)

That's fine don't worry! They'll just fallback to the English translation

> * "Invertible" (in "Image options") - I don't see it in app on my tablet. Could you tell me, what does it do, please?

This option is only shown if you have "Invert notes in dark mode" enabled in Settings. If an image is invertible, it will be darkened/inverted when a note is inverted

**Comment by ceskyDJ** [comment]
Ok, thanks for your notes. I'll try to complete the translation using them.

**Comment by ceskyDJ** [comment]
Ok, Czech hasn't any term for "DIRTY" in this meaning, I think. So, I use something like "test version", because it is close to it, isn't it?

**Comment by ceskyDJ** [comment]
> That's fine don't worry! They'll just fallback to the English translation
Is there a option to add translation for these two items?

**Comment by adil192** [comment]
> Ok, Czech hasn't any term for "DIRTY" in this meaning, I think. So, I use something like "test version", because it is close to it, isn't it?

Yeah that sounds good

> > That's fine don't worry! They'll just fallback to the English translation
> 
> Is there a option to add translation for these two items?

Oh I thought you meant you didn't know how to translate some strings. In this case, it's something built into Flutter. I believe Flutter automatically translates them

**Comment by adil192** [comment]
Let me know when you're ready to merge!

File: lib/i18n/strings_cs.i18n.json [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    }, {
+      "q": "K čemu je heslo pro šifrování?",
+      "a": "Heslo pro šifrování je použito k zašifrování vašich dat před jejich odesláním na server. Není uloženo na serveru a je použito pouze aplikací Saber pro zašifrování a dešifrování dat.\n\nBez jeho znalosti nemůže nikdo přistoupit k vašim poznámkám uloženým na serveru. To také znamená, že pokud ho zapomenete, ztratíte přístup ke svým datům."
+    }, {
+      "q": "Proč používat dvě hesla?",
+      "a": "Heslo k NextCloud účtu se používá k přístupu do cloudu. Heslo pro šifrování \"utají\" vaše data ještě před tím, než se na cloud dostanou.\n\nI když by někdo získat přístup k vašemu NextCloud účtu, vaše poznámky zůstanou v bezpečí zašifrované vlastním heslem. To vám poskytuje druhou vrstvu ochrany vašich dat."
+    }]
+  },
+  "appInfo": {
+    "licenseNotice": "Saber  Copyright (C) 2022  Adil Hanney\nTento program je poskytován bez jakékoliv záruky. Jedná se o software poskytovaný zdarma, který je možné šířit při splnění daných podmínek.",
```
Comment (Luciogi) [inline lib/i18n/strings_cs.i18n.json:98]:
```suggestion
    "licenseNotice": "Saber  Copyright (C) 2023  Adil Hanney\nTento program je poskytován bez jakékoliv záruky. Jedná se o software poskytovaný zdarma, který je možné šířit při splnění daných podmínek.",
```

Comment (adil192) [reply lib/i18n/strings_cs.i18n.json:98]: Good catch! I'll change this to 2022-yyyy (where yyyy is the year of the latest release) after this PR gets merged

Comment (ceskyDJ) [reply lib/i18n/strings_cs.i18n.json:98]: Oh, yes 😅. I just copied what I found for English version.

**Comment by ceskyDJ** [comment]
> Yeah that sounds good, or maybe something like "prerelease" or "beta"

I don't think so. Your "release" is beta right now (completely without guarantee and in active development, I hope you know what I mean).

Prerelease is another problematic term. We use something like "before production release" for translating this. Maybe it is more clear but it is more "advanced" term in Czech, so many people don't understand it.

I would let it be like it is. It is pretty good.

**Comment by ceskyDJ** [comment]
> Oh I thought you meant you didn't know how to translate some strings. In this case, it's something built into Flutter. I believe Flutter automatically translates them

No, these are very simple for transportation 🙂. Ok, so it isn't my business.

**Comment by ceskyDJ** [comment]
I have one another problem - I can't test the transition without changing my device's system language (English right now). It is a time consuming task and every time I do that something breaks down. Could you add manual language switching (not just for this case), please? I think sometimes the user has better knowledge of different language than English, so when his/her language isn't supported, he/she could choose it. Someone uses e.g. English as a system language (due to voice assistant support, for example) but is more comfortable to use his/her native language in apps, when it is available.

File: lib/i18n/strings_cs.i18n.json (hunk 1 above)
Comment (ceskyDJ) [reply lib/i18n/strings_cs.i18n.json:98]: Btw, could I change the date to interval now or you need it in another PR?

Comment (adil192) [reply lib/i18n/strings_cs.i18n.json:98]: No don't worry I'll handle that. It needs some minor changes to the code to make it automatically update

Comment (ceskyDJ) [reply lib/i18n/strings_cs.i18n.json:98]: Ok, so changes are ready to be merged.

**Comment by adil192** [comment]
> Could you add manual language switching (not just for this case), please?

Please can you create an issue for this feature request?

In the meantime, if you have Android 13, you can choose a [per-app language](https://developer.android.com/guide/topics/resources/app-languages).

**Comment by ceskyDJ** [comment]
> Please can you create an issue for this feature request?

Of course.

> In the meantime, if you have Android 13, you can choose a [per-app language](https://developer.android.com/guide/topics/resources/app-languages).

Ok, I didn't know about it. I have newer Android (than 10) only for a few weeks :). Thanks for letting know.

**Comment by adil192** [comment]
Ready to merge?

**Comment by adil192** [comment]
Hi, would you be interested in being mentioned when there's new strings that need to be translated? https://github.com/adil192/saber/issues/19#issuecomment-1306096416

**Comment by ceskyDJ** [comment]
> Hi, would you be interested in being mentioned when there's new strings that need to be translated? [#19 (comment)](https://github.com/adil192/saber/issues/19#issuecomment-1306096416)

Yes, you can tag me in these cases.

**Pull request merged**